plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/functions/tablechanges/TableChangesFunction.java

```diff
@@ -0,0 +1,170 @@
+    private static final String TABLE_NAME_ARGUMENT = "TABLE_NAME";
+    private static final String SINCE_VERSION_ARGUMENT = "SINCE_VERSION";
+    private static final String CHANGE_TYPE_COLUMN_NAME = "_change_type";
+    private static final String COMMIT_VERSION_COLUMN_NAME = "_commit_version";
+    private static final String COMMIT_TIMESTAMP_COLUMN_NAME = "_commit_timestamp";
+    private static final Set<String> IGNORED_COLUMNS = ImmutableSet.of(PATH_COLUMN_NAME, FILE_SIZE_COLUMN_NAME, FILE_MODIFIED_TIME_COLUMN_NAME);
+
+    private final TrinoFileSystemFactory fileSystemFactory;
+    private final DeltaLakeMetadataFactory deltaLakeMetadataFactory;
+
+    public TableChangesFunction(TrinoFileSystemFactory fileSystemFactory, DeltaLakeMetadataFactory deltaLakeMetadataFactory)
+    {
+        super(
+                SCHEMA_NAME,
+                NAME,
+                ImmutableList.of(
+                        ScalarArgumentSpecification.builder().name(TABLE_SCHEMA_NAME_ARGUMENT).type(VARCHAR).build(),
+                        ScalarArgumentSpecification.builder().name(TABLE_NAME_ARGUMENT).type(VARCHAR).build(),
+                        ScalarArgumentSpecification.builder().name(SINCE_VERSION_ARGUMENT).type(BIGINT).defaultValue(-1L).build()),
+                GENERIC_TABLE);
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "fileSystemFactory is null");
+        this.deltaLakeMetadataFactory = requireNonNull(deltaLakeMetadataFactory, "deltaLakeMetadataFactory is null");
+    }
+
+    @Override
+    public TableFunctionAnalysis analyze(ConnectorSession session, ConnectorTransactionHandle transaction, Map<String, Argument> arguments)
+    {
+        ScalarArgument schemaNameArgument = (ScalarArgument) arguments.get(TABLE_SCHEMA_NAME_ARGUMENT);
+        checkFunctionArgument(schemaNameArgument.getValue() != null, "schema_name value for function table_changes() cannot be null");
+        String schemaName = ((Slice) schemaNameArgument.getValue()).toStringUtf8();
+
+        ScalarArgument tableNameArgument = (ScalarArgument) arguments.get(TABLE_NAME_ARGUMENT);
+        checkFunctionArgument(tableNameArgument.getValue() != null, "table_name value for function table_changes() cannot be null");
+        String tableName = ((Slice) tableNameArgument.getValue()).toStringUtf8();
+
+        ScalarArgument sinceVersionArgument = (ScalarArgument) arguments.get(SINCE_VERSION_ARGUMENT);
+        Long sinceVersion = (Long) sinceVersionArgument.getValue() + 1; //+1 to ensure that the sinceVersion is exclusive
+        checkFunctionArgument(sinceVersion >= 0, "since_version value for function table_changes() cannot be negative");
+
+        SchemaTableName schemaTableName = new SchemaTableName(schemaName, tableName);
+        DeltaLakeMetadata deltaLakeMetadata = deltaLakeMetadataFactory.create(session.getIdentity());
+        DeltaLakeTableHandle tableHandle = deltaLakeMetadata.getTableHandle(session, schemaTableName);
```

Comment:
Note that the table may not exist

https://github.com/trinodb/trino/blob/198785f588f6d54046da0a532c44e552f30c28b2/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java#L412-L415

Can you please add a test case (if not already added) for the situation where the schema/table  is missing ?